test/wallet-http-test.js

```diff
@@ -1601,6 +1602,478 @@ describe('Wallet HTTP', function() {
+            address: addr,
+            value: value
+          }]
+        });
+      }
+
+      const blockConnects = common.forEvent(wdb, 'block connect', 1);
+      await mineBlocks(1, w1addr);
+      await blockConnects;
+    };
+
+    const checkDoubleSpends = (txs) => {
+      const spentCoins = new BufferSet();
+
+      for (const tx of txs) {
+        for (const input of tx.inputs) {
+          const key = input.prevout.toKey();
+
+          if (spentCoins.has(key))
+            throw new Error(`Input ${input.prevout.format()} is already spent.`);
+
+          spentCoins.add(key);
+        }
+      }
+    };
+
+    const wMineBlocks = async (n = 1) => {
+      const forConnect = common.forEvent(wdb, 'block connect', n);
+      await mineBlocks(n, w1addr);
+      await forConnect;
+    };
+
+    before(async () => {
+      w1addr = (await wallet.createAddress('default')).address;
+      const winfo1 = await wclient.createWallet(WNAME1, {
+        passphrase: PASSPHRASE1
+      });
+
+      const winfo2 = await wclient.createWallet(WNAME2, {
+        passphrase: PASSPHRASE2
+      });
+
+      assert(winfo1);
+      assert(winfo2);
+
+      // Fund primary wallet.
+      await wMineBlocks(5);
+    });
+
+    beforeEach(async () => {
+      await rcwallet1.lock();
+      await rcwallet2.lock();
+    });
+
+    it('should fund 3 new transactions', async () => {
+      const promises = [];
+
+      await fundNcoins(rcwallet1, 3);
+
+      const forMemTX = common.forEvent(node.mempool, 'tx', 3);
+
+      for (let i = 0; i < 3; i++) {
+        promises.push(rcwallet1.send({
+          passphrase: PASSPHRASE1,
+          subtractFee: true,
+          hardFee: HARD_FEE,
+          outputs: [{
+            address: w1addr,
+            value: FUND_VALUE
+          }]
+        }));
+      }
+
+      const results = await Promise.all(promises);
+      const txs = results.map(details => MTX.fromHex(details.tx));
+      checkDoubleSpends(txs);
+
+      await forMemTX;
+      await wMineBlocks(1);
+
+      const balance = await rcwallet1.getBalance();
+
+      assert.strictEqual(balance.confirmed, 0);
+      assert.strictEqual(balance.unconfirmed, 0);
+      assert.strictEqual(balance.coin, 0);
+    });
+
+    it('should open 3 name auctions', async () => {
+      await fundNcoins(rcwallet1, 3);
+
+      for (let i = 0; i < 3; i++)
+        NAMES.push(rules.grindName(10, node.chain.tip.height, network));
+
+      const promises = [];
+
+      const forMemTX = common.forEvent(node.mempool, 'tx', 4);
+
+      for (let i = 0; i < 3; i++) {
+        promises.push(rcwallet1.createOpen({
+          name: NAMES[i],
+          passphrase: PASSPHRASE1,
+          hardFee: HARD_FEE
+        }));
+      }
+
+      const results = await Promise.all(promises);
+      const txs = results.map(result => MTX.fromHex(result.hex));
+      checkDoubleSpends(txs);
+
+      // spend all money for now.
+      await rcwallet1.send({
+        subtractFee: true,
```

Comment:
is passphrase needed here for consistency?

Reply:
This works because the wallet is unlocked at this stage. (Wallet is locked in the afterEach)
I will just add comment.